pkg/ingester/ingester.go

```diff
@@ -1799,7 +1823,7 @@ func (i *Ingester) executeChunksQuery(ctx context.Context, db *userTSDB, from, t
 	// It's not required to return sorted series because series are sorted by the Mimir querier.
 	ss := q.Select(ctx, false, hints, matchers...)
 	if ss.Err() != nil {
-		return 0, 0, ss.Err()
+		return 0, 0, errors.Wrap(ss.Err(), "iterating ChunkSeriesSet")
```

Comment:
Is this message correct? The error didn't occur while `iterating ChunkSeriesSet` but while finding matching series.

Reply:
Good point, since it is possible that `q.Select` would return an iterator that fails immediately, even before attempting to iterate. It doesn't make much practical difference, since according to iterator protocol, the iterator might not fail until you've called `Next`.